torch/distributions/kl.py

```diff
@@ -278,6 +279,31 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    if p.batch_shape != q.batch_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different batch shapes cannot be computed")
+
+    def _batch_trace(bmat):
+        mat_size = bmat.size(-1)
+        values = torch.stack([torch.trace(M) for M in bmat.contiguous().view((-1, mat_size, mat_size))])
+        return values.view(bmat.shape[:-2])
+
+    def _batch_mm(bmat1, bmat2):
+        mat_size = bmat1.size(-1)
+        bmat1_squash = bmat1.contiguous().view((-1, mat_size, mat_size))
```

Comment:
I believe `.reshape(-1, n, n)` is now preferred over `.contiguous().view(-1, n, n)`, but I'm not sure.

Reply:
There was discussion on Slack about this, and I also think the `.precision_matrix` PR did the same. Will modify accordingly.